windows/devices/uwb/include/windows/devices/uwb/UwbConnector.hxx

```diff
@@ -226,15 +221,52 @@ private:
     std::vector<std::shared_ptr<::uwb::UwbRegisteredDeviceEventCallbacks>>
     GetResolvedDeviceEventCallbacks();
 
+    /**
+     * @brief Internal helper function to insert the callback into the hashmap of ids to callbacks
+     * You MUST have grabbed the m_eventCallbacksGate mutex prior to calling this.
+     *
+     * @param callback
+     * @return std::weak_ptr<::uwb::RegisteredCallbackToken>
+     */
+    std::shared_ptr<::uwb::RegisteredCallbackToken>
+    InsertDeviceEventCallback(std::weak_ptr<::uwb::UwbRegisteredDeviceEventCallbacks> callback);
+
+    /**
+     * @brief Internal helper function to insert the callback into the hashmap of ids to callbacks
+     * You MUST have grabbed the m_eventCallbacksGate mutex prior to calling this.
+     *
+     * @param callback
+     * @return std::weak_ptr<::uwb::RegisteredCallbackToken>
+     */
+    std::shared_ptr<::uwb::RegisteredCallbackToken>
+    InsertSessionEventCallback(std::weak_ptr<::uwb::UwbRegisteredSessionEventCallbacks> callback);
+
+    /**
+     * @brief Internal helper function that deregisters event callbacks.
+     * If you pass in a token that is no longer valid, this function does nothing
+     * You must have grabbed the m_eventCallbacksGate mutex prior to calling this
+     *
+     * @param token
+     */
+    void
+    DeregisterEventCallback(::uwb::RegisteredCallbackToken token);
+
 private:
     // the following shared_mutex is used to protect access to both session and device event callbacks
     mutable std::shared_mutex m_eventCallbacksGate;
-    std::unordered_map<uint32_t, std::vector<std::weak_ptr<::uwb::UwbRegisteredSessionEventCallbacks>>> m_sessionEventCallbacks;
-    std::vector<std::weak_ptr<::uwb::UwbRegisteredDeviceEventCallbacks>> m_deviceEventCallbacks;
+    std::unordered_map<uint32_t, std::vector<uint32_t>> m_sessionEventCallbacks;
+    std::vector<uint32_t> m_deviceEventCallbacks;
     std::string m_deviceName{};
     std::jthread m_notificationThread;
     wil::shared_hfile m_notificationHandleDriver;
     OVERLAPPED m_notificationOverlapped;
+
+private:
+    // the following is related to how the UwbConnector keeps track of its callbacks
+    uint32_t m_tokenUniqueState;
+    std::unordered_map<uint32_t, std::weak_ptr<::uwb::UwbRegisteredSessionEventCallbacks>> m_sessionEventCallbacksIdMap;

```

Comment:
I think the names may be switched from the members above. This is called `m_sessionEventCallbacksIdMap` but stores the actual callbacks, while above you have `m_sessionEventCallbacks` which is a map that stores the callback IDs.

Reply:
m_sessionEventCallbacksIdMap stores the map of the id to the actual callback. I agree that m_sessionEventCallbacks is a misnomer, so I'll rename it